src/CSSList/CSSBlockList.php

```diff
@@ -23,13 +23,23 @@ abstract class CSSBlockList extends CSSList
     /**
      * Gets all `DeclarationBlock` objects recursively, no matter how deeply nested the selectors are.
      *
-     * @return array<int, DeclarationBlock>
+     * @return list<DeclarationBlock>
      */
     public function getAllDeclarationBlocks(): array
     {
-        /** @var array<int, DeclarationBlock> $result */
+        /** @var list<DeclarationBlock> $result */
         $result = [];
-        $this->allDeclarationBlocks($result);
+
+        foreach ($this->contents as $directSibling) {
```

Comment:
Sibling of whom?  Direct (usu. adjacent) sibling (e.g. CSS `+`) normally refers to two of them.  I think the original `$item` was fine.